phylogenetic/config/auspice_config_all.json

```diff
@@ -1,7 +1,8 @@
 {
   "title": "Real-time tracking of dengue virus evolution",
   "maintainers": [
-    {"name": "Trevor Bedford", "url": "http://bedford.io/team/trevor-bedford/"}

```

Comment:
These 10 files seem very similar and seem ripe to be dynamically produced by a snakemake rule. It'd make changes such as this commit a one-liner. This is [what we do in ncov](https://github.com/nextstrain/ncov/blob/master/workflow/snakemake_rules/export_for_nextstrain.smk#L128-L330).

Reply:
I concur that having 10 auspice config files seems like unnecessary duplication. Thank you for providing the link to the rule in ncov. While I can decipher the rule, I find it not very easy to read. 

Would you be open to using an "auspice_config_template.json" file instead? (commit: https://github.com/nextstrain/dengue/pull/18/commits/ca0a3f2eb7b17462b224109f099bc83af3942925) This template file could contain text placeholders like `REPLACE_CLADE_KEY` and `REPLACE_GENE`, which can be replaced using a shell `sed` command.

This separate template.json file might be easier for users to edit, as opposed to explaining the python+json in the ncov rule's run statement. However, I'm undecided on this and open to hearing your thoughts.

Reply:
Yeah, I'm not wedded to any particular approach. In your current implementation of `sed` however it's important to properly escape the strings you are adding as replacement strings.

Reply:
After getting bitten by sed and awk many times myself in these kinds of workflows, I'd recommend an approach that uses a Python script and/or templating language to derive Auspice config JSONs from a template. It might be worth looking into [Snakemake's built-in templating support](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#template-rendering-integration), too.

Reply:
> While I can decipher the rule, I find it not very easy to read.

The ncov example has a lot more complexity than would be needed here. You would be able to write something like:

```py
run:
  data = {
    "title": f"Real-time tracking of dengue virus {wildcards.gene} evolution",
    "maintaners": [],
    "etc": "etc"
  }
  with open(output.auspice_config, 'w') as fh:
    json.dump(data, fh, indent=2)
```

Reply:
Thanks, @jameshadfield, that was more straightforward then I anticipated. https://github.com/nextstrain/dengue/pull/18/commits/fc6c8375deedebf2c0b201e0773cd676c4326a83

Also, thanks to @huddlej for the Snakemake template hint and docs (supported Snakemake 7 onward). I'll explore the template method as I'm curious for potential future applications.

Reply:
This dynamic generation of auspice config files was split out into a separate PR and has since been merged.
* https://github.com/nextstrain/dengue/pull/45

Thanks all!